lib/src/ga.dart

```diff
@@ -9,17 +9,26 @@ import 'dart:js';
 /// Very lightweight Google Analytics integration. This class depends on having
 /// the JavaScript GA library available.
 class Analytics {
+  String _snippetId = "";
+
   Analytics();
 
   bool get isAvailable => _gaFunction != null;
 
   void sendPage() => _ga2('send', 'pageview');
 
+  // A unique ID for embedded DartPads. These are used to distinguish between
+  // instances in the context of an article or codelab.
+  set snippetId(String id) {
+    _snippetId = id;
+  }
+
   void sendEvent(String category, String action, {String label}) {
     Map m = {
       'hitType': 'event',
       'eventCategory': category,
-      'eventAction': action
+      'eventAction': action,
+      'snippetId': _snippetId,
```

Comment:
cc/ @RedBrogdon @galeyang @jayoung-lee I'm not 100% sure this is what we need. Can you take a look?

Reply:
Gale and Ja Young are the authorities here, but IIRC we had been talking about using virtual pageviews to set the URL for subsequent events:

https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications

Reply:
While I don't mind how the data is sent to GA, I wonder how the `snippetId` in this code works. Is it going to be recorded as an "event label"? Or is this a [custom dimension](https://support.google.com/analytics/answer/2709828?hl=en)?